controllers/machinedeployment_sync.go

```diff
@@ -113,8 +113,14 @@ func (r *MachineDeploymentReconciler) getNewMachineSet(ctx context.Context, d *c
 		annotationsUpdated := mdutil.SetNewMachineSetAnnotations(d, msCopy, newRevision, true, log)
 
 		minReadySecondsNeedsUpdate := msCopy.Spec.MinReadySeconds != *d.Spec.MinReadySeconds
-		if annotationsUpdated || minReadySecondsNeedsUpdate {
+		deletePolicyNeedsUpdate := d.Spec.Strategy.RollingUpdate.DeletePolicy != nil && msCopy.Spec.DeletePolicy != *d.Spec.Strategy.RollingUpdate.DeletePolicy
```

Comment:
It looks like this field isn't currently being defaulted in `api/v1alpha4/machinedeployment_webhook.go`, if it's defaulted to the appropriate value there, then this comparison can be simplified to `msCopy.Spec.DeletePolicy != *d.Spec.Strategy.RollingUpdate.DeletePolicy`

Reply:
This was discussed [here](https://github.com/kubernetes-sigs/cluster-api/issues/2585#issuecomment-607500483), and we planned on defaulting to the machineset's default deletepolicy.

Reply:
Yeah either that, or if we want to be explicit, we should have one defaulting function for both, so it's consistent across the two

Reply:
@detiber do you think I should change to add default for md delete policy as well, or leave it as it is?